File: Data Structures & Algorithms/count-subsequences/submission-1.cpp
class Solution {
public:
    int numDistinct(string s, string t) {
       vector<vector<int>> dp(t.length(), vector<int>(s.length()));

       for(int i=0;i<t.length();i++) {
            for(int j=0;j<s.length();j++) {
                if(j!=0) dp[i][j] += dp[i][j-1];
                dp[i][j] += (t[i]==s[j])*(i!=0&&j!=0? dp[i-1][j-1]:(i==0));
            }
       }

       return dp[t.length()-1][s.length()-1];
    }
};

File: Data Structures & Algorithms/greatest-common-divisor-of-strings/submission-1.java
class Solution {
    private int gcd(int a, int b) {
        return a%b==0 ? b : gcd(b, a%b);
    }

    public String gcdOfStrings(String str1, String str2) {
        String str = str1+str2;
        String rev = str2+str1;
        if(!str.equals(rev)) return "";
        return str1.substring(0, gcd(str1.length(), str2.length()));    
    }
}